### PR TITLE
auth: pdnsutil check-zone: DNAME fixes

### PR DESCRIPTION
**Pull request by Habbie**
### Short description
* don't warn about occlusion of records with the same name as the DNAME record
* do warn about occlusion when the DNAME is at the apex

### Checklist

I have:
- [x] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [x] compiled this code
- [x] tested this code
- [ ] included documentation (including possible behaviour changes)
- [ ] documented the code
- [ ] added or modified regression test(s)
- [ ] added or modified unit test(s)
- [ ]  checked that this code was merged to master

### CODE REVIEW

File: pdns/pdnsutil.cc [hunk 1]
```diff
@@ -641,6 +644,11 @@ static int checkZone(DNSSECKeeper &dk, UeberBackend &B, const DNSName& zone, con
         continue;
       }
       if( rr.qname.isPartOf( qname.first ) ) {
+        if( qname.second == QType::DNAME && rr.qname == qname.first ) {
```
Comment (zeha) [inline pdns/pdnsutil.cc:647]: can't say these if/if/if/if blocks are very readable anymore. took me quite some time to understand why this is a new case over the first if. i don't understand however why this is inside the rr.qname.isPartOf bit.

Comment (Habbie) [reply pdns/pdnsutil.cc:647]: I see now that 3 lines up also mentions DNAME. I can probably do better.

Comment (Habbie) [reply pdns/pdnsutil.cc:657]: combining the ifs would not have made things more readable. Instead I added a bunch of comments.

**Comment by zeha** [comment]
The new comments are much appreciated, thanks!

**Pull request merged**